[BUG] GPU get_json_object returns None for quoted special character keys in JSON path in dataproc 2.2

**Describe the bug**
Build: rapids-it-dataproc-2.2-ubuntu22/200 (26.02.0)

The GPU implementation of `get_json_object` fails when parsing JSON paths containing quoted special characters. Test `test_get_json_object_quoted_question` shows that extracting a value using path `$['?']` from JSON `{"?":"QUESTION"}` returns `None` on GPU while CPU correctly returns `'QUESTION'`. This is a functional regression in GPU JSON path parsing.

Error logs:
```
[2026-02-15T13:56:54.139Z] =================================== FAILURES ===================================
[2026-02-15T13:56:54.139Z] _____________________ test_get_json_object_quoted_question _____________________
[2026-02-15T13:56:54.139Z] 
[2026-02-15T13:56:54.139Z] def test_get_json_object_quoted_question():
[2026-02-15T13:56:54.139Z]     schema = StructType([StructField("jsonStr", StringType())])
[2026-02-15T13:56:54.139Z]     data = [[r'{"?":"QUESTION"}']]  
[2026-02-15T13:56:54.139Z]     assert_gpu_and_cpu_are_equal_collect(
[2026-02-15T13:56:54.139Z]         lambda spark: spark.createDataFrame(data,schema=schema).select(
[2026-02-15T13:56:54.139Z]             f.get_json_object('jsonStr',r'''$['?']''').alias('question')))
[2026-02-15T13:56:54.139Z] 
[2026-02-15T13:56:54.139Z] E AssertionError: GPU (None) and CPU (QUESTION) string values are different at [0, 'question']
[2026-02-15T13:56:54.139Z] --- CPU OUTPUT
[2026-02-15T13:56:54.139Z] +++ GPU OUTPUT
[2026-02-15T13:56:54.139Z] -Row(question='QUESTION')
[2026-02-15T13:56:54.139Z] +Row(question=None)
```

**Environment details**
Dataproc 2.2-ubuntu22 (release/26.02)

--- a/integration_tests/src/main/python/get_json_test.py
+++ b/integration_tests/src/main/python/get_json_test.py
@@ -19,7 +19,7 @@ from data_gen import *
 from pyspark.sql.types import *
 from marks import *
 from spark_init_internal import spark_version
-from conftest import is_dataproc_runtime
+from conftest import is_dataproc_runtime, is_dataproc_serverless_runtime
 from spark_session import is_before_spark_400, is_databricks113_or_later, is_databricks_runtime
 
 def mk_json_str_gen(pattern):
@@ -123,7 +123,7 @@ def test_get_json_object_normalize_non_string_output():
             f.col('jsonStr'),
             f.get_json_object('jsonStr', '$')))
 
-@pytest.mark.xfail(condition=is_dataproc_runtime(),
+@pytest.mark.xfail(condition=is_dataproc_runtime() or is_dataproc_serverless_runtime(),
     reason="https://github.com/NVIDIA/spark-rapids/issues/14290")
 def test_get_json_object_quoted_question():
     schema = StructType([StructField("jsonStr", StringType())])

